pkg/component/component.go

```diff
@@ -526,19 +526,87 @@ func ValidateComponentCreateRequest(client *occlient.Client, componentSettings c
 //	appName: Name of application of which the component is a part
 //	componentName: Name of the component which is being patched with config
 //	componentConfig: Component configuration
+//  	cmpExist: true if components exists in the cluster
 // Returns:
 //	err: Errors if any else nil
-func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo, context string, stdout io.Writer) (err error) {
+func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo, stdout io.Writer, cmpExist bool) (err error) {
+	log.Successf("Applying component settings to component: %v", componentConfig.GetName())
+	// if component exist then only call the update function
+	if cmpExist {
 
-	if err = Update(client, componentConfig, componentConfig.GetSourceLocation(), stdout); err != nil {
+		if err = Update(client, componentConfig, componentConfig.GetSourceLocation(), stdout); err != nil {
+			return err
+		}
+	}
+	// URL creation part
+	err = ApplyConfigCreateUrl(client, componentConfig)
+	if err != nil {
+		return err
+	}
+	// URL deletion part
```

Comment:
1 minor comment but I think it'll be nicer to move everything related to url into 1 function and call that here bcoz this function for example would be expected to handle other resources also so more lines of code for each resources implies  complexity of this function increases as a multiple of no.of lines per resource

Reply:
i.e, if ApplyConfigCreateURL can probably be called ApplyURLConfig and andle all creations, deletions etc...

Reply:
fixed

Reply:
It doesn't seem to be fixed I was talking about moving applyConfigDeleteUrl below into ApplyConfigCreateUrl or ApplyURLConfig(as I suggested)